src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/CSharpVirtualDocumentTest.cs

```diff
@@ -98,5 +98,27 @@ public void Update_MultipleEdits()
             var text = textBuffer.CurrentSnapshot.GetText();
             Assert.Equal("Replaced", text);
         }
+
+        [Fact]
+        public void Update_NoChanges_InvokesPostChangedEventTwice_NoEffectiveChanges()
+        {
+            // Arrange
+            var textBuffer = new TestTextBuffer(new StringTextSnapshot("Hello World"));
+            var called = 0;
+            textBuffer.PostChanged += (s, a) =>
+            {
+                called += 1;
+            };
+
+            var document = new CSharpVirtualDocument(Uri, textBuffer);
+
+            // Act
+            document.Update(Array.Empty<TextChange>(), hostDocumentVersion: 1);
+
+            // Assert
+            Assert.Equal(2, called);
```

Comment:
Is it necessary to assert on the `hostDocumentVersion`?

Reply:
Not necessary as we aren't testing that specifically but it definitely doesn't hurt 👍